pkgs/applications/science/electronics/dsview/default.nix

```diff
@@ -0,0 +1,89 @@
+{ stdenv, fetchFromGitHub, pkgconfig, cmake, autoreconfHook,
+glib, libzip, boost, libserialport, fftw, qtbase,
+python3, check, libusb, libftdi, libudev, alsaLib,
```

Comment:
`libudev` is not an available package. It currently is an alias to `udev` which is an alias to `systemd`, but you can't use the alias, you'd need to put it as `systemd` if you want to use it. Are you sure you need it? Looking at `libsigrok4dsl` in AUR, they don't use it: https://aur.archlinux.org/packages/libsigrok4dsl/

Reply:
I started as a fork from the pulseview package. You are right it's not used, I will remove it.

Reply:
I checked and it is indeed [a dependency](https://github.com/DreamSourceLab/DSView/blob/b31f432a4765e88784772d3416721436bce6dbe4/libsigrok4DSL/configure.ac#L256) of libsigrok4dsl.

What is odd tough is that even if I completly remove it, the `./configure` step still seems to find it:
```
Detected libraries:
[...]
  - libudev >= 151: yes (239)
[...]
```

I tought that is exactly what nix should prevent? Or is systemd kind of a available everywhere by default?

Also another question I don't understand is: Why can't the alias not be used here? I think libudev makes is much clearer what is used as a dependency than systemd. Especially if we think about potentially spliting a package in the future. Am I missing something here?

Reply:
You are right, the `libusb` pulls in `libusb1` to your build through `propagatedBuildInputs`, latter pulls in `systemd`. I think, it would be fine to provide `systemd` as an explicit dependency for `libsigrok4DSL`.

Reply:
I will do that.